common/http.go

```diff
@@ -283,16 +283,11 @@ func (h *HTTPClient) sendRequest(req *http.Request) (*http.Response, []byte, err
 		return nil, nil, fmt.Errorf("error reading response body: %w", err)
 	}
 
-	// Check the response status code
-	if res.StatusCode < 200 || res.StatusCode > 299 {
```

Comment:
Moved this logic to the error handler func as I want, to check for errors, even when status code is 200.